src/main/java/de/tum/in/www1/artemis/repository/base/ArtemisJpaRepository.java

```diff
@@ -40,4 +41,21 @@ public interface ArtemisJpaRepository<T, ID> extends JpaRepository<T, ID> {
      * @return the entity with the given id
      */
     T findByIdElseThrow(ID id);
+
+    /**
+     * Find an entity by its id and given specification without using limiting internally.
+     *
+     * @param spec the specification to apply
+     * @param id   the id of the entity to find
+     * @return the entity with the given id
+     */
+    Optional<T> findOneById(Specification<T> spec, ID id);
```

Comment:
Maybe add to the comment that `spec` should only be used for fetching relations. Otherwise there might be unexpected not founds

Reply:
I know it will work, but using filtering specifications doesn't make much sense if you search for a specific ID. Does it?